packages/components/src/text-control/index.js

```diff
@@ -65,6 +66,7 @@ function TextControl(
 				ref={ ref }
 				{ ...props }
 			/>
+			{ children }
```

Comment:
By way of explanation, rendering `children` here was a trade off. 

Following the design requirements, we need to be able to add a dropdown menu as a sibling of the text control input.

We could have created a custom component using the constituent parts of `<TextControl />`. The Advanced panel, however, uses `<TextControl />` multiple times, which means that updates would have to be copied over to the custom component to retain consistency. 

Does thing contravene any component guidelines from a technical or philosophical standpoint? 

As far as I can tell, there are no side effects to adding and using the `children` prop here, though I am worried that it might be "impure" in that it invites manifestations of a base component.

Reply:
I'd be grateful for your thoughts here @ciampo whenever you get time. 🙏

Reply:
Thank you for the ping! Always happy to advise / give direction on components :)

I would say that rendering `children` inside a `TextControl` goes a bit against the way the component is supposed to work. `TextControl`, in my mind, is supposed to be an enriched `input` field.

I would personally recommend that, instead of adding `children` to `TextInput`, you render the dropdown menu as a sibling of `TextInput` in the `CustomClassNameControl` component.

Also, cc'ing @diegohaz and @mirka in case they wanted to add their views here as well

Reply:
Thanks for the feedback and for confirming what the tiny voice in my brain was trying to tell me. I was truly struggling with the conflict 😆 

> I would personally recommend that, instead of adding children to TextInput, you render the dropdown menu as a sibling of TextInput in the CustomClassNameControl component.

I'll try that out, cheers! If the other Advanced control inputs end up looking or working differently I can try to switch them all over to TextInput as well.

Reply:
Ah, I just realized that `TextInput` isn't a component yet. Or at least we haven't migrated it from g2.

I should know, I started it and [never finished](https://github.com/WordPress/gutenberg/pull/32381)! 😅

Reply:
> Ah, I just realized that TextInput isn't a component yet. Or at least we haven't migrated it from g2.

We did initially look into it as well after you started it, but we later stopped because it was a very complicated task, since `TextInput` is very complex (it tries to handle single/multi line text and number input) and represents a very different approach to what we currently have in Gutenberg.

Reply:
> We did initially look into it as well after you started it, but we later stopped because it was a very complicated task

Thanks for the background info. 🙇 

I discovered personally how tricky it was 😬  so I feel better that folks with a lot more experience than I have also felt the same!